src/logic/signupLogic.ts

```diff
@@ -76,21 +75,56 @@ export const signupLogic = kea({
                 setEmail: (_, { email }: { email: string }) => email,
             },
         ],
+        experimentVariants: [
+            {} as Record<string, boolean>,
+            { persist: true },
+            {
+                setVariants: (state: Record<string, boolean>, { newVariants }: { newVariants: string[] }) => {
+                    const nextState = {} as Record<string, boolean>
+                    const existingVariants = Object.keys(state)
+                    newVariants.forEach((variant) => {
+                        if (existingVariants.includes(variant)) {
+                            // Don't overwrite an existing value
+                            nextState[variant] = state[variant]
+                        } else {
+                            nextState[variant] = false
```

Comment:
qq: Why are we defaulting this to false?

Reply:
we load all possible options into state with a default value of `false`, and then choosing to make **one and only one** value `true` when we later detect that all values are `false`.

This is because we need to consider the case that we might turn "off" a variant via disabling its feature flag through our cloud interface. This will result in the variant being removed from state when we check for feature flags on the client in subsequent loads.

So, an easy way to do this in both cases
1. setting one variant for the first time; and
2. setting one variant when the active variant was turned off
is to NOT overwrite existing values, set them all to false and enable only one.